ui/src/frontend/track_panel.ts

```diff
@@ -191,56 +191,77 @@ class TrackShell implements m.ClassComponent<TrackShellAttrs> {
     if (!this.attrs) {
       return;
     }
-    let y = this.attrs.track.getHeight();
-    const mouseMoveEvent = (evMove: MouseEvent): void => {
-      if (!this.attrs) {
-        return;
-      }
-      evMove.preventDefault();
-      let movementY = evMove.movementY;
-      if (this.attrs.pinnedCopy !== true &&
-          isPinned(this.attrs.trackState.id)) {
-        movementY /=2;
-      }
-      y += movementY;
-        if (this.defaultHeight) {
-          const newMultiplier = y / this.defaultHeight;
-          if (newMultiplier < 1) {
-            this.attrs.trackState.scaleFactor = 1;
-          } else {
-            this.attrs.trackState.scaleFactor = newMultiplier;
+    if (e.currentTarget instanceof HTMLElement) {
```

Comment:
On the surface, the logic/code for this (a track) looks the same as for a track group.  And the code involved is non-trivial. I.e., seems like a good amount of code duplication. Could you extract this into some common code that both panel classes use?